standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreListenerNotifier.java

```diff
@@ -209,29 +210,32 @@ public void notify(MetaStoreEventListener listener, ListenerEvent event) throws
               (listener, event) -> listener.onCreateCatalog((CreateCatalogEvent)event))
           .put(EventType.DROP_CATALOG,
               (listener, event) -> listener.onDropCatalog((DropCatalogEvent)event))
-          .put(EventType.OPEN_TXN, new EventNotifier() {
-            @Override
-            public void notify(MetaStoreEventListener listener, ListenerEvent event) throws MetaException {
-              listener.onOpenTxn((OpenTxnEvent)event);
-            }
-          })
-          .put(EventType.COMMIT_TXN, new EventNotifier() {
-            @Override
-            public void notify(MetaStoreEventListener listener, ListenerEvent event)
-                throws MetaException {
-              listener.onCommitTxn((CommitTxnEvent) event);
-            }
-          })
-          .put(EventType.ABORT_TXN, new EventNotifier() {
-            @Override
-            public void notify(MetaStoreEventListener listener, ListenerEvent event)
-                throws MetaException {
-              listener.onAbortTxn((AbortTxnEvent) event);
-            }
-          })
           .build()
   );
 
+
+  private interface TxnEventNotifier {
```

Comment:
Instead of separate interface, can we add a method in EventNotifier itself and shall have default implementation to throw exception if not overridden? This new method shall be overridden only by txn events.